store/tikv/gcworker/gc_worker.go

```diff
@@ -624,27 +642,51 @@ func (w *GCWorker) sendUnsafeDestroyRangeRequest(ctx context.Context, startKey [
 	}
 
 	var wg sync.WaitGroup
+	errChan := make(chan error, len(stores))
 
 	for _, store := range stores {
 		address := store.Address
 		storeID := store.Id
 		wg.Add(1)
 		go func() {
 			defer wg.Done()
-			_, err1 := w.store.GetTiKVClient().SendRequest(ctx, address, req, tikv.UnsafeDestroyRangeTimeout)
+
+			resp, err1 := w.store.GetTiKVClient().SendRequest(ctx, address, req, tikv.UnsafeDestroyRangeTimeout)
+			if err1 == nil {
+				if resp == nil || resp.UnsafeDestroyRange == nil {
+					err1 = errors.New("[gc worker] unsafe destroy range returns nil response")
+				} else {
+					errStr := resp.UnsafeDestroyRange.Error
+					if len(errStr) > 0 {
+						err1 = errors.New("[gc worker] unsafe destroy range failed: " + errStr)
+					}
+				}
+			}
+
 			if err1 != nil {
-				logutil.Logger(ctx).Error("[gc worker] destroy range on store failed with error",
+				metrics.GCUnsafeDestroyRangeFailuresCounter.Inc()
+				logutil.Logger(ctx).Error("[gc worker] unsafe destroy range failed",
 					zap.String("uuid", w.uuid),
 					zap.Uint64("storeID", storeID),
-					zap.Error(err))
-				err = err1
+					zap.Error(err1))
 			}
+			errChan <- err1
 		}()
 	}
 
+	for range stores {
+		err1 := <-errChan
+		if err1 != nil {
+			err = err1
+		}
+	}
+
 	wg.Wait()
 
-	return errors.Trace(err)
+	if err != nil {
+		return errors.New("[gc worker] destroy range failed on some stores")
```

Comment:
Prefer to user `errors.Annotatef` to keep the original error message.

Reply:
These errors were already logged so I thought it's ok to drop them... And there may be multiple regions returned from above, so I think collect them into an `[]error` and use `errors.Errorf`. It's that ok?

Reply:
The error log will be scattered throughout the file in the current implementation. I think we should collect all error messages in a GC job and log them after the job finishes.